src/Markup/Avalonia.Markup.Xaml/XamlIl/Runtime/RuntimePlatformInfo.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Runtime.InteropServices;
+
+
+namespace Avalonia.Markup.Xaml.XamlIl.Runtime;
+
+public class RuntimePlatformInfo
+{
+    private static OSPlatform IOS { get; } = OSPlatform.Create("IOS");
+    
+    private static OSPlatform Android { get; } = OSPlatform.Create("ANDROID");
+    
+    private static OSPlatform Browser { get; } = OSPlatform.Create("BROWSER");
+    
+    public static RuntimePlatformInfo Instance { get; } = new();
+    
+    private RuntimePlatformInfo()
+    {
+    }
+
+    public bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+    public bool IsMacOS => RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+
+    public bool IsLinux => RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+    public bool IsIOS => RuntimeInformation.IsOSPlatform(IOS);
+
+    public bool IsAndroid => RuntimeInformation.IsOSPlatform(Android);
+
+    public bool IsBrowser => RuntimeInformation.IsOSPlatform(Browser);
+
+    public bool IsDesktop(Size primaryScreenSize) => primaryScreenSize.Width > 1280;
```

Comment:
There is no primary screen or any screen when xaml is loading. it's simply _not a concept_ in XAML.